### PR TITLE
♻️ Maintenance: Enhanced logs on sms service errors and tests image labels

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Various improvements:
- ♻️ Adds user_id (in extras) and masked phone number for sms failures @odeimaiz @sanderegg 
   - addresses the issue with failing sms in US 
- ♻️ Adds more validation checks for image-labels in registry, e.g. version in metadata vs image tag @mguidon 
   - addresses issue with wrong version of s4l
- 🔨 Minor improvement in release form

## Related issue/s

- maintenance

## How to test

- test_2fa_sms_failure_during_login
```cmd
cd services/web/server
make install-dev
pytest -vv tests/unit/**/test_*_login.py
```

## DevOps 

None

### CODE REVIEW

File: packages/models-library/tests/test_utils_service_io.py [hunk 1]
```diff
@@ -82,3 +85,37 @@ def test_against_service_metadata_configs(metadata_path: Path):
...
+
+def _iter_main_services() -> Iterable[Path]:
+    """NOTE: Filters the main service when there is a group
+    of services behind a node.
+    """
+    for p in TEST_DATA_FOLDER.rglob("metadata-*.json"):
+        with suppress(Exception):
+            meta = json.loads(p.read_text())
+            if (meta.get("type") == "computational") or meta.get(
+                "service.container-http-entrypoint"
```
Comment (mguidon) [inline packages/models-library/tests/test_utils_service_io.py:103]: :+1:

File: packages/models-library/tests/test_utils_service_io.py [hunk 2]
```diff
@@ -82,3 +85,37 @@ def test_against_service_metadata_configs(metadata_path: Path):
...
+    ids=lambda p: f"{p.parent.name}/{p.name}",
+)
+def test_service_metadata_has_same_version_as_tag(metadata_path: Path):
+    meta = json.loads(metadata_path.read_text())
+
+    # metadata-M.m.b.json
+    match = _VERSION_SEARCH_RE.search(metadata_path.name)
+    assert match, f"tag {metadata_path.name} is not a version"
+    version_in_tag = match.group()
+    assert meta["version"] == version_in_tag
```
Comment (mguidon) [inline packages/models-library/tests/test_utils_service_io.py:121]: :+1:

**Review by mguidon — APPROVED** [review]
Thanks for the test!

File: services/web/server/src/simcore_service_webserver/login/_constants.py [hunk 3]
```diff
@@ -1,6 +1,9 @@
 from typing import Final
 
 MSG_2FA_CODE_SENT: Final[str] = "Code sent by SMS to {phone_number}"
+MSG_2FA_UNAVAILABLE_OEC: Final[
+    str
+] = "Currently we cannot use 2FA, please try again later ({error_code})"
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/login/_constants.py:6]: make it impersonal
```suggestion
] = "2FA cannot be currently used, please try again later ({error_code})"
```

**Review by GitHK — APPROVED** [review]
👍

File: services/web/server/tests/unit/with_dbs/03/login/test_login_2fa.py [hunk 4]
```diff
@@ -322,3 +326,56 @@ async def test_send_email_code(
...
+            # Expects failure:
+            #    HTTPServiceUnavailable: Currently we cannot use 2FA, please try again later (OEC:140558738809344)
+            data, error = await assert_status(response, web.HTTPServiceUnavailable)
+            assert not data
+            assert error["errors"][0]["message"].startswith(
+                MSG_2FA_UNAVAILABLE_OEC[:10]
+            )
+
+            # Expects logs like 'Failed while setting up 2FA code and sending SMS to 157XXXXXXXX3 [OEC:140392495277888]'
+            assert f"{fake_user_phone_number[:3]}" in caplog.text
```
Comment (sanderegg) [inline services/web/server/tests/unit/with_dbs/03/login/test_login_2fa.py:381]: :+1:

**Review by sanderegg — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/login/_constants.py (hunk 3 above)
Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/login/_constants.py:6]: app-teams' message ... delicate to change ;-)

**Review by matusdrobuliak66 — COMMENTED** [review]
:+1:

**Pull request merged**